workflow/snakemake/Snakefile

```diff
@@ -0,0 +1,85 @@
+# Note that if you are working on the analysis development locally, i.e. outside
+# of the REANA platform, you can proceed as follows:
+#
+#   $ cd reana-demo-cms-h4l
+#   $ mkdir snakemake-local-run
+#   $ cd snakemake-local-run
+#   $ virtualenv ~/.virtualenvs/cms-h4l-snakemake
+#   $ source ~/.virtualenvs/cms-h4l-snakemake/bin/activate
+#   $ pip install snakemake
+#   $ cp -a ../code ../data .
+#   $ snakemake -s ../workflow/snakemake/Snakefile -p --cores 4 --use-singularity
+
+
+
+rule all:
+    input:
+        "results/mass4l_combine_userlvl3.pdf"
+
+
+rule analyze_data:
+    input:
+        "data",
+        "code"
+    output:
+        "results/DoubleMuParked2012C_10000_Higgs.root"
+    container:
+        "docker://cmsopendata/cmssw_5_3_32"
+    shell:
+        "source /opt/cms/cmsset_default.sh "
+        "&&  scramv1 project CMSSW CMSSW_5_3_32 "
+        "&& cd CMSSW_5_3_32/src "
+        "&& eval `scramv1 runtime -sh` "
+        "&& cp -r ../../code/HiggsExample20112012 . "
+        "&& cd HiggsExample20112012/HiggsDemoAnalyzer "
+        "&& scram b "
+        "&& cd ../Level3 "
+        "&& mkdir -p ../../../../results "
+        "&& cmsRun demoanalyzer_cfg_level3data.py"
+
+
+
+rule analyze_mc:
+    input:
+        "data",
+        "code"
+    output:
+        "results/Higgs4L1file.root"
+    container:
+        "docker://cmsopendata/cmssw_5_3_32"
+    shell:
+       "source /opt/cms/cmsset_default.sh "
+        "&&  scramv1 project CMSSW CMSSW_5_3_32 "
+        "&& cd CMSSW_5_3_32/src "
+        "&& eval `scramv1 runtime -sh` "
+        "&& cp -r ../../code/HiggsExample20112012 . "
+        "&& cd HiggsExample20112012/HiggsDemoAnalyzer "
+        "&& scram b "
+        "&& cd ../Level3 "
+        "&& mkdir -p ../../../../results "
+        "&& cmsRun demoanalyzer_cfg_level3MC.py"
+
+
+rule make_plot:
+    input:
+        "data",
+        "code",
+        mc_data="results/DoubleMuParked2012C_10000_Higgs.root",
+        a_data="results/Higgs4L1file.root"
+    output:
+        "results/mass4l_combine_userlvl3.pdf"
+    container:
+        "docker://cmsopendata/cmssw_5_3_32"
+    shell:
+        "source /opt/cms/cmsset_default.sh "
+        "&&  scramv1 project CMSSW CMSSW_5_3_32 "
+        "&& cd CMSSW_5_3_32/src "
+        "&& eval `scramv1 runtime -sh` "
+        "&& cp -r ../../code/HiggsExample20112012 . "
+        "&& cd HiggsExample20112012/HiggsDemoAnalyzer "
+        "&& scram b "
+        "&& cd ../Level3 "
+        "&& mkdir -p ../../../../results "
+        "&& cp {input.mc_data} ../../../../results "
```

Comment:
The input ROOT files for the plots are well generated in the "results" directory, so you can delete these two `cp` statements. The workflow would then work; I can get the plots :smile: 

```diff
diff --git a/workflow/snakemake/Snakefile b/workflow/snakemake/Snakefile
index bdf881c..b653f4d 100644
--- a/workflow/snakemake/Snakefile
+++ b/workflow/snakemake/Snakefile
@@ -80,6 +80,4 @@ rule make_plot:
         "&& scram b "
         "&& cd ../Level3 "
         "&& mkdir -p ../../../../results "
-        "&& cp {input.mc_data} ../../../../results "
-        "&& cp {input.a_data} ../../../../results "
         "&& root -b -l -q ./M4Lnormdatall_lvl3.cc"
```